base/experimental.jl

```diff
@@ -660,7 +660,7 @@ function wait_with_timeout(c::GenericCondition; first::Bool=false, timeout::Real
             # don't do anything here.
             if !waiter_left[] && ct.queue == c.waitq
```

Comment:
Won't this `==` also dispatch?

Reply:
Hmm, it might. I'll give the scheduler a once over